src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -4184,8 +4184,17 @@ internal static void AddEventSource(EventSource newEventSource)
         // such callbacks on process shutdown or appdomain so that unmanaged code will never
         // do this.  This is what this callback is for.
         // See bug 724140 for more
+#if ES_BUILD_STANDALONE
         private static void DisposeOnShutdown(object? sender, EventArgs e)
+#else
+        internal static void DisposeOnShutdown()
+#endif
         {
+            if (!EventSource.IsSupported)
```

Comment:
Convert to Assert? It is checked at the callsite already.

Reply:
Sure